src/ripple/ledger/impl/View.cpp

```diff
@@ -757,286 +757,77 @@ dirAdd (ApplyView& view,
-        sleNode->setFieldH256 (sfRootIndex, dir.key);
-        view.insert (sleNode);
-
-        if (uNodeDir != 1)
-            sleNode->setFieldU64 (sfIndexPrevious, uNodeDir - 1);
-
-        fDescriber (sleNode);
+        if (iter.page().key != pageId.key)
+        {
+            if (pageNum++ == 20)
+                break;
+            pageId = iter.page();
+        }
 
-        svIndexes   = STVector256 ();
+        if (iter.index() == item)
+            return pageNum;
     }
 
-    svIndexes.push_back (uLedgerIndex); // Append entry.
-    sleNode->setFieldV256 (sfIndexes, svIndexes);   // Save entry.
-
-    JLOG (j.trace()) <<
-        "dirAdd:   creating: root: " << to_string (dir.key);
-    JLOG (j.trace()) <<
-        "dirAdd:  appending: Entry: " << to_string (uLedgerIndex);
-    JLOG (j.trace()) <<
-        "dirAdd:  appending: Node: " << strHex (uNodeDir);
-
-    return { tesSUCCESS, false };
+    return boost::none;
 }
 
 // Ledger must be in a state for this to work.
 TER
 dirDelete (ApplyView& view,
     const bool                      bKeepRoot,      // --> True, if we never completely clean up, after we overflow the root node.
-    const std::uint64_t&            uNodeDir,       // --> Node containing entry.
+    std::uint64_t                   uNodeDir,       // --> Node containing entry.
     uint256 const&                  uRootIndex,     // --> The index of the base of the directory.  Nodes are based off of this.
     uint256 const&                  uLedgerIndex,   // --> Value to remove from directory.
     const bool                      bStable,        // --> True, not to change relative order of entries.
     const bool                      bSoft,          // --> True, uNodeDir is not hard and fast (pass uNodeDir=0).
     beast::Journal j)
 {
-    std::uint64_t uNodeCur = uNodeDir;
-    SLE::pointer sleNode =
-        view.peek(keylet::page(uRootIndex, uNodeCur));
-
-    if (!sleNode)
-    {
-        JLOG (j.warn()) << "dirDelete: no such node:" <<
-            " uRootIndex=" << to_string (uRootIndex) <<
-            " uNodeDir=" << strHex (uNodeDir) <<
-            " uLedgerIndex=" << to_string (uLedgerIndex);
-
-        if (!bSoft)
-        {
-            assert (false);
-            return tefBAD_LEDGER;
-        }
-        else if (uNodeDir < 20)
-        {
-            // Go the extra mile. Even if node doesn't exist, try the next node.
-
-            return dirDelete (view, bKeepRoot,
-                uNodeDir + 1, uRootIndex, uLedgerIndex, bStable, true, j);
-        }
-        else
-        {
-            return tefBAD_LEDGER;
-        }
-    }
-
-    STVector256 svIndexes = sleNode->getFieldV256 (sfIndexes);
-
-    auto it = std::find (svIndexes.begin (), svIndexes.end (), uLedgerIndex);
-
-    if (svIndexes.end () == it)
+    // Emulate the old semantics: if uNodeDir isn't hard
+    // and fast and there's no page number, search for the
+    // item:
+    if (bSoft && uNodeDir == 0)
```

Comment:
This code searches the first 20 pages of a directory, but only when `uNodeDir == 0`. The code this replaces would search from the current page up to the 20th page (i.e. old code would search even if `uNodeDir` was not zero).